### ISSUE TITLE
[FEATURE]: Sharing a folder to a user

### ISSUE THREAD

**Issue by CypLef** [marius-wieschollek/passwords#583]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **feature request** for _one_ feature, not a question, discussion or multiple features.
- [X] This is not a feature request for the [browser extension](https://github.com/marius-wieschollek/passwords-webextension) or another client.
- [X] This feature is **not** already requested on Github _(I've searched it)_.
- [X] This feature does not already exist _(I checked the [wiki](https://git.mdns.eu/nextcloud/passwords/-/wikis/home))_.

### Current Status

It would be really nice to be able to share an entire folder to a user (like nextcloud folder sharing).

### Feature Description

1. I open the passwords app
2. I click on the folder i want to share ![image](https://user-images.githubusercontent.com/92938361/222490977-577affd4-1819-4bc0-b9b1-4d5a170e0dcc.png)
3. A "sharing" tab is visible on the right panel
4. The same user interface as for sharing a password appear:  ![image](https://user-images.githubusercontent.com/92938361/222491295-114693dc-deea-4492-ab84-4df1afe2ae82.png)
5. I type an user name
6. i select it
7. It share the entire password folder to this user (with rights applying to every password in this folder (editing, re-sharing...)

Later (after the original folder share)

- If i add a password to this folder, this password is automatically shared to the user with whom i shared the folder
- Same if i edit a password
- Same if the user edit/add/remove a password into this folder (it is automatically shared with me) -> Only if i give the editing right to this user.

**Comment by underdoeg** [comment]
I think that is a duplicate of https://github.com/marius-wieschollek/passwords/issues/27

**Comment by osm-frasch** [comment]
In the same sense as this feature request here, it would also make sense to share the corresponding tag when sharing a password.

**Comment by thisiszeev** [comment]
I would like to say a big yes to this feature, also being able to share to a group as a whole is useful. We have many devs working on the same projects, and it is important for us to share various passwords for test accounts, etc.

**Comment by mitchell-accipio** [comment]
@marius-wieschollek  first of all, thank you for all the fine work so far in developing this plugin for Nextcloud. I actually raised this as an example of great open-source (in tandem with Nextcloud) at a talk for London Youth last month.

That being said, this feature has been long desired and is actually a big blocker for me in my personal deployment, since I want to be able to share creds with my family, so I am looking into a potential option to put some paid dev time (not me, not a dev) into this feature; 

It would be helpful for me to understand what about this feature is difficult and what the challenges are that have prohibited it being implemented by the group of contributors so far? I can see that this is a feature lacking in several other open-source password vault solutions, but is available in paid solutions, so my assumption is there is a technical blocker to this. If you could help me understand briefly what that is, it would be very helpful in my task to seek development on this feature. 

Thank you for your time in advance.

**Comment by oleole39** [comment]
> If you could help me understand briefly what that is, it would be very helpful in my task to seek development on this feature.

@mitchell-accipio Those comments could help you (although partially adressing that issue together with the "sharing to group" issue):
- https://github.com/marius-wieschollek/passwords/issues/27#issuecomment-369304957
- https://github.com/marius-wieschollek/passwords/issues/27#issuecomment-1816570324
- https://github.com/marius-wieschollek/passwords/issues/27#issuecomment-372009580

I haven't digged in the code base, but if 1 to 1 sharing is already working, the recursion idea mentioned in the last listed comment for X to 1 sharing seems to be a way to do without having to take big decisions.

Additionally, maybe you'll want to have a chat (although that invitation is already 1-y old):
- https://github.com/marius-wieschollek/passwords/issues/27#issuecomment-1772166679

**Comment by thisiszeev** [comment]
@oleole39 I don't think recursion would be the way to go as it can be come resource intensive for large groups. Added, if a user is removed from the group, it is a painful exercise to have a hook that then removes them from the share. Nextcloud has a good API for working with Group Access to functionality and features. I haven't played with it myself, but I did see it in the developer documentation.

**Comment by Medius7** [comment]
I would love to join in. 
Such a feature would be ideal and wonderful, especially for our collaborative work in a nonprofit organization. For teams already working with Nextcloud, this would be the easiest and best method to share passwords (in "shared NC passwords folders"). It could prevent the insecure sending of passwords via email/messenger or the more cumbersome sharing of a KeePass file. The NC Passwords solution would also have advantages when people leave the team, as it allows access to be revoked easily.

**Comment by easyservices** [comment]
> ### ⚠️ This issue respects the following points: ⚠️
> 
>     * [x]  This is a **feature request** for _one_ feature, not a question, discussion or multiple features.[x]  This is not a feature request for the [browser extension](https://github.com/marius-wieschollek/passwords-webextension) or another client.[x]  This feature is **not** already requested on Github _(I've searched it)_.[x]  This feature does not already exist _(I checked the [wiki](https://git.mdns.eu/nextcloud/passwords/-/wikis/home))_.
> 
> 
> ### Current Status
> 
> It would be really nice to be able to share an entire folder to a user (like nextcloud folder sharing).
> ### Feature Description
> 
>     1. I open the passwords app
> 
>     2. I click on the folder i want to share ![image](https://user-images.githubusercontent.com/92938361/222490977-577affd4-1819-4bc0-b9b1-4d5a170e0dcc.png)
> 
>     3. A "sharing" tab is visible on the right panel
> 
>     4. The same user interface as for sharing a password appear:  ![image](https://user-images.githubusercontent.com/92938361/222491295-114693dc-deea-4492-ab84-4df1afe2ae82.png)
> 
>     5. I type an user name
> 
>     6. i select it
> 
>     7. It share the entire password folder to this user (with rights applying to every password in this folder (editing, re-sharing...)
> 
> 
> Later (after the original folder share)
> 
>     * If i add a password to this folder, this password is automatically shared to the user with whom i shared the folder
> 
>     * Same if i edit a password
> 
>     * Same if the user edit/add/remove a password into this folder (it is automatically shared with me) -> Only if i give the editing right to this user.
> 
> 
> ### Additional Context
> 
> _No response_

Yes! I would be great it this requested feature is implemented. It would be a big improvement for collaborative use, within a family or a team. Password sharings is not convenient when you need to share a complete set of passwords.

**Comment by kabirferro** [comment]
UP >> It would be very useful to be able to share a folder with a group of users, especially in enterprise and team-based environments.

**Comment by goofy79** [comment]
I totally agree. Our team is facing the same challenge. Manually sharing dozens of credentials with every new colleague is prone to errors. Implementing shared folders or ‘Team Vaults’ would ensure everyone stays up-to-date automatically. This is a must-have feature for business users!

**Comment by ssergio-ll** [comment]
I also vote in favor of this functionality. For us, it is essential. Until this feature is available, we are currently using KeeWeb. Sharing each password individually with every team member is operationally chaotic.

**Comment by JohannesRegner** [comment]
We also have more and more customers who use Nextcloud, and they also want to use the password manager. Therefore, this feature is actually essential.

**Comment by kezakis** [comment]
I'd also like to join the submitted request for implementing a shared Nextcloud Passwords folder, as it would greatly improve our team's security and workflow. As a governmental organisation, we require robust and compliant solutions. Please add me as a participant so I can contribute to the discussion and implementation. Thank you!

**Comment by RamonP12** [comment]
> [@marius-wieschollek](https://github.com/marius-wieschollek) first of all, thank you for all the fine work so far in developing this plugin for Nextcloud. I actually raised this as an example of great open-source (in tandem with Nextcloud) at a talk for London Youth last month.

If you could help me understand briefly what that is, it would be very helpful in my task to seek development on this feature.

@mitchell-accipio did you find a defeloper who could help you to develop the "share folder with passwords inside to another User"?
